compiler/src/iree/compiler/Dialect/Flow/IR/FlowTypes.h

```diff
@@ -54,29 +54,30 @@ class DispatchTensorType
  public:
   using ImplType = detail::DispatchTensorTypeStorage;
 
-  static constexpr int64_t kDynamicSize = -1;
-
   using Base::Base;
 
   /// Get or create a new DispatchTensorType of the provided shape and
   /// element type. Assumes the arguments define a well-formed
   /// DispatchTensorType.
   static DispatchTensorType get(TensorAccess access, ArrayRef<int64_t> shape,
-                                Type elementType);
+                                Type elementType, Attribute encoding = {});
 
-  static DispatchTensorType get(TensorAccess access, TensorType tensorType);
+  static DispatchTensorType get(TensorAccess access, Type type);
 
   static DispatchTensorType parse(AsmParser &parser);
 
   /// Returns the allowed operations the tensor.
   TensorAccess getAccess() const;
 
-  /// Return the element type.
-  Type getElementType() const;
+  /// Returns the embedded type.
+  Type getEmbedType() const;
```

Comment:
This got me thinking about the name of the type a bit - effectively these are pointers with special semantics and it's probably the right time to rename them and help clear up the behavior for people - at this level of IR they're (effectively) just fat pointers with access permissions that (later on) get type-erased. Another advantage would be allowing us to point at different types in the future if we wanted (custom user types, image samplers, etc).

The upstream dialects use `LLVM::LLVMPointerType`/`spirv::PointerType` as the C++ type, `llvm.ptr`/`spv.ptr` as the assembly name, and call the inner member the pointee (`getPointeeType()` etc).

Following that we'd have `DispatchPointerType`/`flow.dispatch.ptr`/embed->pointee here.

If you wanted to keep this smaller I'd say just do embed->pointee, and then the DispatchTensorType->DispatchPointerType could come later. I'm happy to do that if you'd like - not confusing new people with dispatch tensor semantics is really motivating :)

Reply:
Pointer/Pointee type can get people to come with some baggage.... but I think it makes sense. If I can get this PR working, I will do the embed -> pointee type change with this PR... DispatchTensorType -> DispatchPointerType must be a regex replace too... if it is more complicated than that, ill leave it to you.

Reply:
Yeah I was debating that too - I call them "bindings" in the stream/hal dialects (as they are in GPU terminology) and we could do that here too for consistency (`flow.dispatch.binding`/maybe `getStorageType`) - that has less baggage as most people coming in with a bias probably don't know what a binding is :)
You're the other half of the people that hear about the naming/semantics of these things so which would you rather explain? I'm fine with either :)

Reply:
yeah, that might be better. So `flow.dispatch.binding` and `getBoundType` ? I fixed my build failures, so I can start with regex replace of the lit tests...